### ISSUE TITLE
Preprocess of bounding boxes - pytorch implementation

### ISSUE THREAD

**Issue by won-bae** [junsukchoe/ADL#3]
Hi, great paper and code! It is very helpful to reproduce previous works. Thanks for your dedication. In fact, I've ran into several questions in your pytorch implementation. 

First, shouldn't image_height and width from Line 108 and 109 be swapped since gt_bbox[2] and gt_bbox[3] refer to x and y coordinates, respectively? https://github.com/junsukchoe/ADL/blob/9cb833de7bec57e05787b0a2c616e6ccfb123a86/Pytorch/utils/util_cam.py#L108

Also, is there any particular reason why you subtracted 1 from resize_size?
https://github.com/junsukchoe/ADL/blob/9cb833de7bec57e05787b0a2c616e6ccfb123a86/Pytorch/utils/util_bbox.py#L41

Lastly, were you able to reproduce the same (or similar) performance as the implementation in tensorflow? If there is a discrepancy, how much was it?

Appreciate for your help in advance.

**Comment by junsukchoe** [comment]
Hi won-bae,   

Thanks for your interest in our work. Here are the answers:

> First, shouldn't image_height and width from Line 108 and 109 be swapped since gt_bbox[2] and gt_bbox[3] refer to x and y coordinates, respectively?

You are right. It should be swapped. Because the height and width of the image are the same, we have not found that it is wrong. Thanks!   



> Also, is there any particular reason why you subtracted 1 from resize_size?

Since the index origin = 0, we subtract 1 from `resize_size`.   



> Lastly, were you able to reproduce the same (or similar) performance as the implementation in tensorflow? If there is a discrepancy, how much was it?

Unfortunately, we do not have enough computing resources for testing the pytorch implementation on ImageNet-1k dataset.  For CUB-200-2011 dataset, we have found that there is slight discrepancy in accuracy. Maybe it is because of different framework. We are working on reducing the gap further between the performances between tensorflow and pytorch implementations.   



Actually, pytorch implementation is work in progress. There may be some bugs or errors. Sorry for your inconvenience.

Thanks,
Junsuk

**Comment by won-bae** [comment]
Thank you so much for your quick reply. Hope the pytorch implementation will be updated soon so that more people can utilize your algorithm.
